### PR TITLE
New package: py-mysql-connector

### PR DESCRIPTION
**Pull request by citibeth**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-mysql-connector/package.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+    homepage = "https://github.com/mysql/mysql-connector-python"
+    url      = "https://github.com/mysql/mysql-connector-python/archive/8.0.13.tar.gz"
+    git      = "https://github.com/mysql/mysql-connector-python.git"
+
+    version('8.0.13', sha256='d4c0834c583cdb90c0aeae90b1917d58355a4bf9b0266c16fd58874a5607f9d4')
+
+    # See not on PythonPackage for why this is type='run', not type='build'
+    # Like py-basemap, py-mysql-connector does not depend on py-setuptools
+    depends_on('py-setuptools', type='run')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-mysql-connector/package.py]: I can see why `py-basemap` is `type='run'`, but shouldn't this package be `type=('build', 'run')`? `setuptools` is actually imported at build time:
https://github.com/mysql/mysql-connector-python/blob/master/setup.py#L40

Also missing a dependency on `py-protobuf@3.0.0:'`:
https://github.com/mysql/mysql-connector-python/blob/master/setupinfo.py#L139

File: var/spack/repos/builtin/packages/py-mysql-connector/package.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    no compilation is necessary to run this Python DB API v2.0
+    compliant driver."""
+
+    homepage = "https://github.com/mysql/mysql-connector-python"
+    url      = "https://github.com/mysql/mysql-connector-python/archive/8.0.13.tar.gz"
+    git      = "https://github.com/mysql/mysql-connector-python.git"
+
+    version('8.0.13', sha256='d4c0834c583cdb90c0aeae90b1917d58355a4bf9b0266c16fd58874a5607f9d4')
+
+    # See not on PythonPackage for why this is type='run', not type='build'
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-mysql-connector/package.py]: typo: not -> note?

**Comment by citibeth** [comment]
Requested changes should be resolved.

**Comment by adamjstewart** [comment]
Now it's missing a dependency on setuptools.

**Comment by citibeth** [comment]
Really?  It worked for me.

**Comment by adamjstewart** [comment]
https://github.com/mysql/mysql-connector-python/blob/master/setup.py#L40

**Comment by adamjstewart** [comment]
Optionally we could also override `install` so that it doesn't pass the `--single-version-externally-managed` flag.

**Comment by citibeth** [comment]
> Optionally we could also override `install` so that it doesn't pass the `--single-version-externally-managed` flag.

I have no way to evaluate which way forward would be better.  But so far I'm happy with the patch I found, because it seems to work.

**Pull request merged**